app/views/layouts/application.html.erb

```diff
@@ -0,0 +1,60 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>MediaRanker</title>
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
+
+    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+  </head>
+
+  <body>
+    <header class="app-header__header">
+      <h1>
+        <%= link_to "Media Ranker", root_path, method: :get %>
+        <small>Ranking the Best of Everything</small>
+      </h1>
+
+      <nav class="app-header__nav">
+        <ul class="nav app-header__site-nav-container">
+          <li class="nav-item app-header__nav_item">
+            <%= link_to "View top media", root_path, method: :get %>
+          </li>
+          <li class="nav-item app-header__nav_item">
+            <%= link_to "View all media", works_path, method: :get %>
+          </li>
+          <li class="nav-item app-header__nav_item">
+            <%= link_to "Add a new work", new_work_path, method: :get %>
+          </li>
+          <li class="nav-item app-header__nav_item">
+            <%= link_to "View all users", users_path, method: :get %>
+          </li>
+        </ul>
+        <ul class="nav app-header__user-nav-container">
+          <% if session[:user_id] %>
+            <li class="nav-item app-header__nav_item">
+              <p><%= link_to "Logged in as #{find_user.username}", user_path(session[:user_id]), method: :get %></p>
```

Comment:
The logic here is what could go in the determine_user method, assigning it to an instance variable. 
`@current_user = user_path(session[:user_id])`

Reply:
That way it can be accessible from any controller or view.